sdk/test/common/baseline_circular_buffer.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <cstdint>
+#include <memory>
+#include <mutex>
+#include <vector>
+
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace testing
+{
+/**
+ * A locking circular buffer.
+ *
+ * Used as a baseline in benchmarking.
+ */
+template <class T>
+class BaselineCircularBuffer
+{
+public:
+  explicit BaselineCircularBuffer(size_t max_size) : data_{max_size} {}
+
+  /**
+   * Add an element to the circular buffer.
+   * @param element the element to add
+   * @return true if the element was added successfully
+   */
+  bool Add(std::unique_ptr<T> &element) noexcept { return this->Add(std::move(element)); }
+
+  bool Add(std::unique_ptr<T> &&element) noexcept
+  {
+    std::lock_guard<std::mutex> lock_gaurd{mutex_};
```

Comment:
Nit: gaurd -> guard

Reply:
Trying to cover this automatically with #54.

Reply:
Oops. Fixed.